indico/modules/events/registration/controllers/management/reglists.py

```diff
@@ -0,0 +1,147 @@
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Indico; if not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import unicode_literals
+
+import itertools
+import json
+from uuid import uuid4
+
+from flask import session, request, redirect, jsonify
+from sqlalchemy.orm import joinedload
+
+from indico.core.db import db
+from indico.modules.events.registration.controllers.management import RHManageRegFormBase, RHManageRegistrationBase
+from indico.modules.events.registration.models.items import RegistrationFormItemType, RegistrationFormItem
+from indico.modules.events.registration.models.registrations import Registration, RegistrationData
+from indico.modules.events.registration.models.form_fields import RegistrationFormFieldData
+from indico.modules.events.registration.views import WPManageRegistration
+from indico.web.flask.templating import get_template_module
+from indico.web.flask.util import url_for
+from indico.web.util import jsonify_data
+from MaKaC.common.cache import GenericCache
+
+
+cache = GenericCache('reglist-config')
+
+
+def _get_filters_from_request(regform):
+    filters = {}
+    for field in regform.form_items:
+        if field.is_field and field.input_type in ['radio', 'country']:
+            radioitems = request.form.getlist('radiogroup_{}'.format(field.id))
+            if radioitems:
+                filters[field.id] = radioitems
+    return filters
+
+
+def _filter_registration(query, filters):
+    if not filters:
+        return query
+
+    criteria = [db.and_(RegistrationFormFieldData.field_id == field_id,
+                        RegistrationData.data.op('#>>')('{}').in_(data_list))
+                for field_id, data_list in filters.iteritems()]
+    subquery = (RegistrationData.query
+                .with_entities(db.func.count(RegistrationData.registration_id))
+                .join(RegistrationData.field_data)
+                .filter(RegistrationData.registration_id == Registration.id)
+                .filter(db.or_(*criteria))
+                .correlate(Registration)
+                .as_scalar())
+    return query.filter(subquery == len(filters))
+
+
+def _query_registrations(regform, filters):
+    query = (Registration.query
+             .with_parent(regform)
+             .options(joinedload('data').joinedload('field_data').joinedload('field')))
+    return _filter_registration(query, filters)
+
+
+class RHRegistrationsListManage(RHManageRegFormBase):
+    """List all registrations of a specific registration form of an event"""
+
+    def _process(self):
+        session_key = 'reg_list_config_{}'.format(self.regform.id)
+        report_config_uuid = request.args.get('config')
+        if report_config_uuid:
+            configuration = cache.get(report_config_uuid)
```

Comment:
IIRC. we wanted to include the regform it in the data stored in the cache to avoid loading another regform's config (which would probably not work or even cause errors)